tests/iter.rs

```diff
@@ -25,6 +25,16 @@ fn struct_iter() {
     assert_eq!(paths.next(), None);
 }
 
+#[test]
+fn struct_iter_indices() {
+    let mut paths = [([0, 0], 1), ([1, 0], 1), ([2, 0], 2)].into_iter();
+    for (have, expect) in Settings::iter_indices().zip(&mut paths) {
```

Comment:
It would be a good sanity to check to also add an access of the item via the iterated indices and verify they point to what we think they do.

Reply:
Maybe convert the indices to a path for each and verify they are what we think they should be.